cql3/statements/select_statement.cc

```diff
@@ -694,7 +694,7 @@ select_statement::execute(service::storage_proxy& proxy,
     // is specified we need to get "limit" rows from each partition since there
     // is no way to tell which of these rows belong to the query result before
     // doing post-query ordering.
-    auto timeout = db::timeout_clock::now() + get_timeout(options);
+    auto timeout = db::timeout_clock::now() + get_timeout(state.get_client_state(), options);
```

Comment:
Another way to go would be to set query_options::_timeout_config to point to the client state's _timeout_config. Not saying it's better, just pointing out the alternative.